projects/pgai/tests/vectorizer/cli/test_vectorizer_document.py

```diff
@@ -0,0 +1,476 @@
+        chunks = cur.fetchall()
+        chunks_str = "\n".join([chunk["chunk"] for chunk in chunks])
+
+        # Sacred Texts of PostgreSQL.pdf
+        assert "And lo, there came forth PostgreSQL, blessed be its name" in chunks_str
+
+        # sample_pdf.pdf
+        assert "Maecenas mauris lectus" in chunks_str
+
+        # sample_with_table.pdf
+        assert "This is an example of a data table." in chunks_str
+
+        # sample_txt.txt
+        assert "Fromage frais cheese and biscuits danish fontina" in chunks_str
+
+        # basic-v3plus2.epub
+        # epub is not supported by docling but by pymupdf. auto parser handles this.
+        assert "Wants pawn term dare worsted ladle gull hoe lift" in chunks_str
+
+        # test.md
+        assert "Hello I am a test md document" in chunks_str
+
+        # lego_sets.pdf
+        assert "7190-1 Millennium Falcon" in chunks_str
+
+
+def test_simple_document_embedding_s3_no_credentials(
+    s3_bucket: str,
+    cli_db: tuple[TestDatabase, Connection],
+    cli_db_url: str,
+):
+    """Test that embedding fails when no credentials are available"""
+    vectorizer_id = configure_document_vectorizer(
+        cli_db[1], base_path=f"s3://{s3_bucket}"
+    )
+
+    if "AWS_ACCESS_KEY_ID" in os.environ:
+        del os.environ["AWS_ACCESS_KEY_ID"]
+    if "AWS_SECRET_ACCESS_KEY" in os.environ:
+        del os.environ["AWS_SECRET_ACCESS_KEY"]
+
+    # This is necessary to prevent boto3 from using the default credentials file.
+    os.environ["AWS_SHARED_CREDENTIALS_FILE"] = "/dev/null"
+
+    # No cassette because it should never get to an API call.
+    run_vectorizer_worker(cli_db_url, vectorizer_id)
+
+    connection = cli_db[1]
+    with connection.cursor(row_factory=dict_row) as cur:
+        cur.execute("SELECT count(*) FROM ai._vectorizer_q_1;")
+        assert cur.fetchone()["count"] == 1  # type: ignore
+        cur.execute("SELECT id, message, details FROM ai.vectorizer_errors")
+        records = cur.fetchall()
+        assert len(records) == 1
+        error = records[0]
+        assert error["id"] == vectorizer_id
+        assert error["message"] == "loading failed"
+        assert error["details"] == {
+            "loader": "uri",
+            "error_reason": "Unable to locate credentials",
+            "is_retryable": True,
+        }
+
+
+def test_simple_document_embedding_s3(
+    s3_bucket: str,
+    cli_db: tuple[TestDatabase, Connection],
+    cli_db_url: str,
+    vcr_: Any,
+):
+    """Test that a document is successfully embedded"""
+    connection = cli_db[1]
+    vectorizer_id = configure_document_vectorizer(
+        cli_db[1], base_path=f"s3://{s3_bucket}"
+    )
+
+    # required for accessing this S3 bucket in particular
+    # os.environ["AWS_DEFAULT_REGION"] = "eu-central-1"
+
+    # This is necessary to prevent boto3 from pulling credentials
+    # from WS Instance Metadata Service when those are missing.
+    # Comment if you need to recreate the cassette.
```

Comment:
can we detect whether we are recreating the cassete in code and then programmatically set the right things>